source/dotnet/Library/AdaptiveCards/AdaptiveBackgroundImageConverter.cs

```diff
@@ -0,0 +1,58 @@
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+using System.Reflection;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+
+namespace AdaptiveCards
+{
+    public class AdaptiveBackgroundImageConverter : JsonConverter, ILogWarnings
+    {
+        public List<AdaptiveWarning> Warnings { get; set; } = new List<AdaptiveWarning>();
+
+        public override void WriteJson(JsonWriter writer, object value, JsonSerializer serializer)
+        {
+            AdaptiveBackgroundImage bi = (AdaptiveBackgroundImage) value;
+            if (bi.HasDefaultValues())
+            {
+                writer.WriteValue(bi.BackgroundImageString);
+            }
+            else
```

Comment:
Should this check if BackgroundImageString is null or empty and then not serialize anything?